internal/k8s/container.go

```diff
@@ -120,10 +120,11 @@ func ContainerIDFromContainerStatus(status v1.ContainerStatus) (container.ID, er
 		return "", nil
 	}
 
-	if !strings.HasPrefix(id, ContainerIDPrefix) {
+	components := strings.SplitN(id, "://", 2)
+	if len(components) != 2 {
 		return "", fmt.Errorf("Malformed container ID: %s", id)
 	}
-	return container.ID(id[len(ContainerIDPrefix):]), nil
+	return container.ID(components[1]), nil
```

Comment:
Just out of curiosity - do we really need to drop the prefix or care what this id looks like? does this open us up to conflicts in a cluster that has, e.g., both cri-o and docker? Is it weird to assume that k8s has this prefix there for a reason?

(though this seems pretttty unlikely to bite us, so probably isn't worth spending much time on)

Reply:
I'm not sure if Docker accepts docker://<id> or just <id>. I think you need to split it out before sending it (but could be wrong), but yeah, we could keep it here and split it later, but, *shrug*